app/models/cell.rb

```diff
@@ -1,6 +1,14 @@
 class Cell < ApplicationRecord
   belongs_to :player
   belongs_to :coordinate
-  belongs_to :color
-  belongs_to :pip
+  belongs_to :color, optional: true
+  belongs_to :pip, optional: true
+
+  def diagonals
+    player.cells.where(coordinate: coordinate.neighbour_coordinates.where(neighbours: {direction: 'diagonal'}))
```

Comment:
On the Coordinate model:

def diagonals
   neighbour_coordinates.where(neighbours: {direction: 'diagonal'})
end

On Cell model:

def diagonals
    player.cells.where(coordinate: coordinate.diagonals)
end